### PR TITLE
saltpack: general package documentation

### PR DESCRIPTION
**Pull request by maxtaco**

### CODE REVIEW

**Comment by maxtaco** [comment]
PTAR @oconnor663?

**Comment by oconnor663** [comment]
Please take a rook?

File: go/saltpack/doc.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+Package saltpack is an implementation of the Saltpack message format. Saltpack
```
Comment (oconnor663) [inline go/saltpack/doc.go]: Lowercase "saltpack" here?

File: go/saltpack/doc.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+Saltpack makes no attempt to manage keys. We assume the wrapping application
+has a story for key management.
+
+Modes of Operation
+
+Saltpack supports three modes of operation: encrypted messages, attached
+signatures, and detached signatures. An attached signature contains a message
+and a signature that authenticates it. A detached signature contains just the
+signature, and assumes an independent delievery mechanism for the file
+(this might come up when distributing an ISO an separate signature of it).
```
Comment (oconnor663) [inline go/saltpack/doc.go]: s/an/and a/ ?

File: go/saltpack/doc.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/*
+Package saltpack is an implementation of the Saltpack message format. Saltpack
+is a very light wrapper around Dan Berstein's famous NaCl library. It adds
```
Comment (oconnor663) [inline go/saltpack/doc.go]: I'm not sure I agree with "light wrapper around NaCl". Maybe something more like a "combination of NaCl and MessagePack into a message format for (some) humans."

**Pull request merged**